src/Microsoft.Azure.SignalR.Common/Endpoints/AadAccessKey.cs

```diff
@@ -41,13 +68,43 @@ public async Task AuthorizeAsync(string endpoint, int? port, string serverId, Ca
             return await base.GenerateAccessToken(audience, claims, lifetime, algorithm);
         }
 
-        public Task<string> GenerateAadToken()
+        public async Task UpdateAccessKeyAsync(IServerNameProvider provider, ServiceEndpoint endpoint, ILoggerFactory loggerFactory)
         {
-            if (_authOptions is IAadTokenGenerator options)
+            if (!_initializeTcs.TrySetResult(true))
```

Comment:
a bool is enough?

Reply:
I'm not sure if the function will be called twice or more.
If the `AccessKey` could be shared among different endpoints then it's necessary to use a `TaskCompletionSource` to ensure the idempotency.